sycl/unittests/kernel-and-program/DeviceInfo.cpp

```diff
@@ -64,7 +64,7 @@ TEST_F(DeviceInfoTest, GetDeviceUUID) {
     return;
   }
 
-  context Ctx{Plt};
+  context Ctx{Plt.get_devices()[0]};
```

Comment:
Are we sure there's always more than 1 device? That's probably not the problem of this patch, but rather the problem of the whole test.

Reply:
Do you mean if there's no device in the platform? I believe in that scenario the `default_selector` would throw a runtime exception way before we reach this point.

Reply:
@npmiller I agree, I'm just not sure if we should skip test in this case, or report it as a failure. Or maybe mock some apis, so that we always have a device?

Reply:
Maybe, although since these are in-tree lit tests and they should also be run with a device filter I'm not sure if there really is a use case where they would run without a device available. And I think unless a use case for having them skip shows up it's probably good to have them throw exceptions so they don't silently skip if there's a problem with device selection for example.

Reply:
Unit tests are not filtered anyhow. But that's not a problem of this particular PR.